[oolongwater] iP

oolongwater opened the pull request:
### **GiorgoLite: Your Personal Task Wizard! 🧙‍♂️**

> “I think teeth are the first reminder of mortality. I think that's where it begins to hit ya. Cus they don't come back and that's it.” - Terry A. Davis ([source](https://www.goodreads.com/author/quotes/7171073.Terry_A_Davis))

Giorgo is here to lighten the load on your memory and help you stay on top of your game! It’s:

- Simple to use
- Beautifully minimalistic
- Powerful enough to handle all your tasks effortlessly
- Fast Blazingly fast so you can get things done in a flash ⚡

With Giorgo, you can:

- Type away your tasks in seconds.
- Organize your life like a ~~noob~~ pro.
- Stay focused and ~~stressed~~ stress-free 😊.

Ready to boost your productivity with a touch of magic? 🎩✨

All you need to do is,

- [x] Download the _.jar_ file from [here](https://github.com/oolongwater/ip/releases/tag/A-Jar).
- [x] Copy the _.jar_ file into an empty folder.
- [ ] Open a command window in that folder.
- [ ] Run the command `java -jar "{filename}.jar"` e.g., `java -jar "Duke.jar"` (i.e., run the command in the same folder as the _.jar_ file).

And it is **FREE!**

Features:

1.  Managing tasks
2.  Managing deadlines
3.  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

oolongwater committed:
Level-0

oolongwater committed:
Level-1

oolongwater committed:
Level-2

oolongwater committed:
Level-3

oolongwater committed:
Level-4

oolongwater committed:
A-TextUiTesting

oolongwater committed:
Level-5

oolongwater committed:
Level-6

oolongwater committed:
A-Enums

oolongwater committed:
Level-7

oolongwater committed:
Level-8

oolongwater committed:
Level-8 Stretch

oolongwater committed:
A-MoreOOP

oolongwater committed:
A-Packages

oolongwater committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

oolongwater committed:
Updated delete command

oolongwater committed:
A-Gradle

oolongwater committed:
Deleted extra src

oolongwater committed:
A-JUnit

oolongwater committed:
A-Jar

oolongwater committed:
A-Jar Fixed gradle

oolongwater committed:
A-JavaDoc

oolongwater committed:
Level-9

oolongwater committed:
A-CheckStyle

oolongwater committed:
Level-10 Deep-Fried GUI

oolongwater committed:
Level-10 Normal GUI

oolongwater committed:
A-Varargs

oolongwater committed:
Fixed Date Bug

oolongwater committed:
Add assert statements to document important assumptions

Added assert statements in various parts of the code to document
important assumptions that should hold at various points. This helps
in identifying potential issues early and ensures that the code
behaves as expected.

Changes include:
- Added asserts in `Giorgo.java` to check for non-null tasks and
  non-empty arguments.
- Added asserts in `Event.java` and `Deadline.java` to validate
  file format and structure.
- Added asserts in `Storage.java` to handle unknown task types
  and ensure proper task loading.

These changes improve code robustness and make debugging easier
by catching invalid states early.

oolongwater committed:
Merge pull request #1 from oolongwater/A-Assertions

Add assert statements to document important assumptions

oolongwater committed:
Refactor code for improved readability and maintainability

The codebase contains several areas where readability and maintainability can be improved. This includes adding Javadoc comments, using modern Java features, and refactoring methods for clarity.

To address these issues, the following changes have been made:
* Added Javadoc comments to the `Giorgo` class and its methods to provide detailed descriptions of their functionality.
* Improved code quality in `Storage.java` by using try-with-resources and extracting task creation logic into a separate method.
* Enhanced `Main.java` and `Launcher.java` for better readability and maintainability.
* Refactored the `getTasksOnDate` method in `Ui.java` for clarity and efficiency.

These changes help ensure the code is easier to understand and maintain, reducing the likelihood of bugs and making future enhancements simpler.

oolongwater committed:
Merge pull request #2 from oolongwater/A-CodeQuality

Refactor code for improved readability and maintainability

oolongwater committed:
Use Java Streams for improved code readability and efficiency

The codebase contains several areas where the use of Java Streams can
improve readability and efficiency. This includes refactoring methods to
use Streams for filtering, mapping, and collecting data.

To address these issues, the following changes have been made:

* Refactored the `getTasksOnDate` method to use Java Streams for filtering
  and collecting tasks.
* Refactored the `load` method in the `Storage` class to use Java Streams
  for reading lines from the file and creating tasks.
* Refactored the `save` method in the `Storage` class to use Java Streams
  for writing tasks to the file.

These changes help ensure the code is more concise and easier to
understand, while also leveraging the benefits of Java Streams for
processing collections.

oolongwater committed:
Merge pull request #3 from oolongwater/A-Streams

Use Java Streams for improved code readability and efficiency

oolongwater committed:
Update README.md

oolongwater committed:
Add priority feature to tasks

Tasks in the application did not have a priority level, making it
difficult for users to manage and prioritize their tasks effectively.

Adding a priority feature allows users to assign a priority level to
each task, helping them to better organize and manage their tasks
based on importance.

Let's:
* Modify the `UI` class to display the priority level in the task
  added message.
* Update the `handleTodo`, `handleDeadline`, and `handleEvent` methods
  in `Giorgo.java` to handle the priority level.
* Adjust the `parseDeadlineArguments` method in the `Parser` class to
  correctly parse the deadline input with priority.
* Ensure the `Deadline` class constructor can handle single-digit
  dates and months.

This approach ensures that the priority feature is integrated
seamlessly into the existing task management system, enhancing the
user experience.

oolongwater committed:
Merge tag 'BCD-Extension'

oolongwater committed:
Update README.md

Peanuts359 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,73 @@
+package main.java;
```
I think the GUI logic should be shifted to its own package and kept separate from the core class

oolongwater replied:
Good idea, thanks

Peanuts359 commented on src/task/Task.java:
```diff
@@ -0,0 +1,85 @@
+package task;
```
It's good to see that you moved all your task classes to their own package.

Peanuts359 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,132 @@
+ */
+public class Ui {
+
+    /**
```
You actually went to the point of adding JavaDoc comments to EVERY SINGLE METHOD. Nice.

oolongwater replied:
Thanks!

Peanuts359 reviewed (commented):
Good job on sticking to even some of the advanced coding standard rules.
There was one minor inconsistency in this regard, but other than that, it's good to go.

oolongwater committed:
docs: Add UI screenshot and update command descriptions in README

The README.md file now includes a screenshot of the UI to provide users
with a visual reference. Additionally, the command descriptions have been
updated for better clarity and accuracy.

Let's,
* add the `Ui.png` image to the `docs` directory
* update the README.md to display the image
* refine the command descriptions in the README.md

oolongwater committed:
docs: Update command descriptions in README

The command descriptions have been
updated with the priority feature for better clarity and accuracy.

Let's,
* refine the command descriptions to reflect priority in the README.md

oolongwater committed:
chore(release): prepare for final release

- Ensure all dependencies are up to date
- Verify main class configuration in `build.gradle`
- Confirm correct package structure and classpath
- Update documentation and README

Reallyeasy1 commented on src/storage/Storage.java:
```diff
@@ -0,0 +1,109 @@
+        List<Task> tasks = new ArrayList<>();
+        Path path = Paths.get(filePath);
+
+        try {
```
hello

Reallyeasy1 replied:
May I know why the try catch loop is nested? Is it possible to un-nest the try-catch block?

Reallyeasy1 commented on src/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+        } catch (DateTimeParseException e) {
+            e.printStackTrace();
+        }
+        assert event != null;
```
In the try catch block, if there is DateTimeParseException, event can be null.
The assertion line may raise an exception. Are there any ways to handle this?

Reallyeasy1 replied:
In the try catch block, if there is DateTimeParseException, event can be null. The assertion line may raise an exception. Are there any ways to handle this?

Reallyeasy1 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,184 @@
+                if (task instanceof Deadline deadline) {
+                    if (deadline.getBy().toLocalDate().isEqual(inputDate)) {
+                        sb.append(count).append(") [Priority: ").append(
+                                task.getPriority()).append("] ").append(task).append("\n\n");
```
Are there any ways of un-nesting the if-else nests?

Reallyeasy1 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,184 @@
+                    LocalDate startDate = event.getFrom().toLocalDate();
+                    LocalDate endDate = event.getTo().toLocalDate();
+                    if ((inputDate.isEqual(startDate) || inputDate.isAfter(startDate))
+                            && (inputDate.isEqual(endDate) || inputDate.isBefore(endDate))) {
```
Is it possible to split this into individual boolean values?
(eg. isGTEStartDate and isLTEEndDate)

Reallyeasy1 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,184 @@
+            if (task instanceof Deadline deadline) {
+                if (deadline.getBy().toLocalDate().isEqual(inputDate)) {
+                    sb.append(count).append(") [Priority: ").append(
+                            task.getPriority()).append("] ").append(task).append("\n\n");
```
Are there any ways of un-nesting the if-else nests?

Reallyeasy1 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,184 @@
+            } else if (task instanceof Event event) {
+                LocalDate startDate = event.getFrom().toLocalDate();
+                LocalDate endDate = event.getTo().toLocalDate();
+                if ((inputDate.isEqual(startDate) || inputDate.isAfter(startDate))
```
Are there any ways of un-nesting the if-else nests?

Reallyeasy1 commented on src/ui/Ui.java:
```diff
@@ -0,0 +1,184 @@
+        }
+
+        return sb.toString();
+    }
```
Good job in writing your code, relatively minor changes only.
